parlai/core/agents.py

```diff
@@ -327,6 +327,13 @@ def create_agent(opt, requireModelExists=False):
     the options file if it exists (the file opt['model_file'] + '.opt' must exist and
     contain a pickled dict containing the model's options).
     """
+    if opt.get('datapath', None) is None:
+        # add datapath, it is missing
+        from parlai.core.params import ParlaiParser
+        parser = ParlaiParser(add_model_args=True)
```

Comment:
I think you can change this to just `parser = ParlaiParser()` (don't need the model args)

Reply:
or even `parser = ParlaiParser(add_parlai_args=False); parser.add_parlai_datapath()` since that's all we need